analyses/malawi/notebooks/mwi_ecmwf_monthly_skill_dryspells.md

```diff
@@ -0,0 +1,827 @@
+
+ax.set_xlabel("Monthly rainfall threshold (mm)", labelpad=20, weight='bold', size=20)
+ax.set_ylabel("Percentage", labelpad=20, weight='bold', size=20)
+ax.set_ylim(0,100)
+sns.despine(bottom=True,left=True)
+
+# Draw vertical axis lines
+vals = ax.get_yticks()
+for tick in vals:
+    ax.axhline(y=tick, linestyle='dashed', alpha=0.4, color='#eeeeee', zorder=1)
+
+plt.title(f"Percentage of months that are correctly categorized for the given threshold")
+handles, labels = ax.get_legend_handles_labels()
+fig.legend(handles, labels, loc='upper center')
+fig.tight_layout(rect=(0,0,1,0.9))
+```
+
+```python
+fig,ax=plt.subplots()
+
+df_pr_th.plot(x="threshold",y="month_miss_rate" ,figsize=(16, 8), color='#F2645A',legend=False,ax=ax,style='.-',label="dry spell occurred and monthly precipitation above threshold (misses)")
+df_pr_th.plot(x="threshold",y="month_false_alarm_rate" ,figsize=(16, 8), color='#66B0EC',legend=False,ax=ax,style='.-',label="no dry spell occurred and monthly precipitation below threshold (false alarms)") #["#18998F","#FCE0DE"]
+
+# Set x-axis label
+ax.set_xlabel("Monthly rainfall threshold (mm)", labelpad=20, weight='bold', size=20)
+
+# Set y-axis label
+ax.set_ylabel("Percentage", labelpad=20, weight='bold', size=20)
+
+# Despine
+ax.spines['right'].set_visible(False)
+ax.spines['top'].set_visible(False)
+ax.spines['left'].set_visible(False)
+ax.spines['bottom'].set_visible(False)
+
+ax.set_ylim(0,100)
+
+# Draw vertical axis lines
+vals = ax.get_yticks()
+for tick in vals:
+    ax.axhline(y=tick, linestyle='dashed', alpha=0.4, color='#eeeeee', zorder=1)
+
+plt.title(f"Percentage of months that are correctly categorized for the given threshold")
+handles, labels = ax.get_legend_handles_labels()
+fig.legend(handles, labels, loc='upper center')
+fig.tight_layout(rect=(0,0,1,0.9))
+```
+
+```python
+print("point of intersection")
+df_pr_th[df_pr_th.month_ds>=df_pr_th.month_no_ds].head(1)
+```
+
+```python
+pr_list=[]
+threshold_list=np.arange(0,df_ds_for.mean_cell.max() - df_ds_for.mean_cell.max()%10,10)
+unique_lt=df_ds_for.leadtime.unique()
+
+for m in unique_lt:
+    df_pr_perlt=pd.DataFrame(threshold_list,columns=["threshold"]).set_index(['threshold'])
+    df_ds_for_lt=df_ds_for[df_ds_for.leadtime==m]
+    y_target =  df_ds_for_lt.dry_spell
+    
+    for t in threshold_list:
+        y_predicted = np.where(df_ds_for_lt.mean_cell<=t,1,0)
+
+        cm = confusion_matrix(y_target=y_target, 
+                              y_predicted=y_predicted)
+        tn,fp,fn,tp=cm.flatten()
+        df_pr_perlt.loc[t,["month_ds","month_no_ds","month_miss_rate","month_false_alarm_rate","precision","recall","num_trig","detection_rate"]]=tp/(tp+fn)*100,tn/(tn+fp)*100,fn/(tp+fn)*100,fp/(tp+fp+0.000001)*100,tp/(tp+fp+0.00001)*100,tp/(tp+fn)*100,tp+fp,tp/(tp+fn)*100
```

Comment:
Since you are calculating some of these quantities in other places, it might be good to make some functions to do it, just to avoid any potential errors